src/webapi/src/controller/user_repository.rs

```diff
@@ -134,3 +133,77 @@ pub async fn login_user(
     let token = create_token(pool, &user_id).await?;
     Ok(token)
 }
+
+pub async fn get_user_id_by_token(
+    pg_pool: &PgPool,
+    token: SessionToken,
+) -> Result<Uuid, UserRepositoryError> {
+    let result = sqlx::query!(
+        r#"
+    SELECT user_id, valid_until from sessions 
+    WHERE token = $1
+            "#,
+        token.to_database_value().expose_secret().to_owned()
+    )
+    .fetch_one(pg_pool)
+    .await
+    .map_err(|e| match e {
+        sqlx::Error::RowNotFound => UserRepositoryError::SessionNotFound,
+        _ => {
+            tracing::error!("Failed to fetch session from database: {:?}", e);
+            UserRepositoryError::InternalError
+        }
+    })?;
+
+    if result.valid_until > chrono::Utc::now() {
+        Ok(result.user_id)
+    } else {
+        let _ = delete_session_by_token(pg_pool, token).await;
+        Err(UserRepositoryError::SessionNotFound)
+    }
+}
+
+pub async fn delete_session_by_token(
+    pg_pool: &PgPool,
+    token: SessionToken,
+) -> Result<(), UserRepositoryError> {
+    sqlx::query!(
+        r#"
+    DELETE FROM sessions 
+    WHERE token = $1
+            "#,
+        token.to_database_value().expose_secret().to_owned()
+    )
+    .execute(pg_pool)
+    .await
+    .map_err(|e| {
+        tracing::error!("Failed to delete token from database: {:?}", e);
+        UserRepositoryError::InternalError
+    })?;
```

Comment:
Will sqlx return an error if no rows will be affected? A double logout shouldn't result in `InternalError`

Reply:
Feel free to fix that when adding logout